### ISSUE TITLE
searchFields delimiter

### ISSUE THREAD

**Issue by cdion** [keystonejs/keystone#280]
The documentation states that the searchFields option for a List should be space-separated.  However, this causes keystone to generate an invalid mongoose criteria object when using that List in a relationship and invoking the autocomplete api.

For example:
```
var Users = new keystone.List('users', {
  searchFields: 'firstname lastname email',
  ...
});

var Test = keystone.List(...);
Test.add({
  user: { type: Types.Relationship, ref: 'users' }
});
```

When editing a Test, and selecting a User, the following request to the autocomplete api is generated:
```
/keystone/api/users/autocomplete?q=Chris&limit=10&page=1&context=relationship&list=users&field=user&item=5342d51839d2b60000901a21&_=1396889204554
```

Which calls (in routes/api/list):
```
var filters = req.list.getSearchFilters(req.query.q);
```

In this case, the filter object returned by req.list.getSearchFilters produces this invalid mongoose criteria object (there is no field with that name defined in the User list):
```
{ 'firstname lastname email': /Chris/i }
```

However, when specifying searchFields with delimited by commas, req.list.getSearchFilters produces a working mongoose criteria object, like so:
```
{ '$or': 
   [ { firstname: /Lauren/i },
     { lastname: /Lauren/i },
     { email: /Lauren/i } ] }
```

The simple resolution is to update the documentation to require searchFields to be comma-delimited.  I haven't looked deeper to see what would be required to get req.list.getSearchFilters to work with spaces.

**Comment by JedWatson** [comment]
Closed by keystonejs/keystonejs-site#26

**Comment by bradwbradw** [comment]
Documentation still says "space-delimited":

http://keystonejs.com/docs/database/#lists-options

https://github.com/keystonejs/keystonejs-site/pull/26#issuecomment-158942517
